### PR TITLE
As a user, I can filter search result by a ad count and it's amount

### PR DESCRIPTION
**Pull request by hanam1ni**
## What happened

- Add ad count filter
- Refactor view to partial

## Proof Of Work

Keywords can be filtered with ad count amount.

image

### CODE REVIEW

File: lib/google_crawler_web/templates/keyword/_keyword_filter.html.eex [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<div class="collapse keyword-filter" id="keywordFilter">
+  <div class="card card-body">
+    <form>
+      <div class="form-group row">
+        <div class="keyword-filter__item col-lg-4 col-6">
+          <label for="title">Keyword Title</label>
+          <input class="form-control" name="keyword_filter[title]" placeholder='"apple", "rocket"'
```
Comment (andyduong1920) [inline lib/google_crawler_web/templates/keyword/_keyword_filter.html.eex:7]: Should we remove the double quote in `"apple", "rocket"`

Comment (hanam1ni) [reply lib/google_crawler_web/templates/keyword/_keyword_filter.html.eex:7]: This one already fixed in the base branch. So it should be updated after rebasing 😁

Comment (hanam1ni) [reply lib/google_crawler_web/templates/keyword/_keyword_filter.html.eex]: @andyduong1920 This PR is rebased, the placeholders are fixed 😁

**Review by andyduong1920 — APPROVED** [review]
All good @hanam1ni 👍

File: lib/google_crawler/keywords.ex [hunk 2]
```diff
@@ -72,8 +73,8 @@ defmodule GoogleCrawler.Keywords do
     |> group_by([keyword], keyword.id)
     |> select_merge([search_results: sr], %{
       result_count: count(sr.id),
-      ad_count: fragment("sum(?::int)", sr.is_ad),
-      top_ad_count: fragment("sum(?::int)", sr.is_top_ad)
+      ad_count: fragment("COALESCE(SUM(?::int), 0)", sr.is_ad),
```
Comment (bterone) [inline lib/google_crawler/keywords.ex:76]: Could we somehow use this without fragments like `ad_count: type(sr.is_ad, :integer) |> sum() |> coalesce(0)`? 💭 

Just so I'm understanding it correctly, it's casting the search result bool to an integer, we're getting the sum, and if it's nil it's zero instead? 🤔

**Review by bterone — APPROVED** [review]

File: lib/google_crawler/keywords.ex (hunk 2 above)
Comment (hanam1ni) [reply lib/google_crawler/keywords.ex:76]: @bterone yes, you understand correctly. If the value is nil, we need to make it 0 to make it filterable

And I think for this case using fragment is look simpler than use pipe 🤔

**Pull request merged**